include/executors/blas1_tree_executor.hpp

```diff
@@ -126,13 +126,29 @@ struct Evaluate<ScalarOp<Operator, SCL, RHS>> {
   using input_type = ScalarOp<Operator, SCL, RHS>;
   using type = ScalarOp<Operator, scl_type, rhs_type>;
 
-  static type convert_to(input_type v, cl::sycl::handler& h) {
+  static type convert_to(input_type v, cl::sycl::handler &h) {
     auto scl = Evaluate<SCL>::convert_to(v.scl, h);
     auto rhs = Evaluate<RHS>::convert_to(v.r, h);
     return type(scl, rhs);
   }
 };
 
+/*! Evaluate<TupleOp<Operator, RHS>
+ * @brief See Evaluate.
+ */
+template <typename RHS>
+struct Evaluate<TupleOp<RHS>> {
+  using value_type = typename RHS::value_type;
+  using rhs_type = typename Evaluate<RHS>::type;
+  using input_type = TupleOp<RHS>;
+  using type = TupleOp<rhs_type>;
+
+  static type convert_to(input_type v, cl::sycl::handler &h) {
+    auto rhs = Evaluate<RHS>::convert_to(v.r, h);
+    return type(rhs);
+  }
+};
+
```

Comment:
Evaluate is in all three files, and Evaluate<TupleOP> is an evaluator specific to blas1 interface.